src/sentry/api/endpoints/organization_teams.py

```diff
@@ -142,25 +143,53 @@ def post(self, request: Request, organization, **kwargs) -> Response:
 
         :pparam string organization_slug: the slug of the organization the
                                           team should be created for.
-        :param string name: the optional name of the team.
-        :param string slug: the optional slug for this team.  If
-                            not provided it will be auto generated from the
-                            name.
+        :qparam string name: the optional name of the team.
+        :qparam string slug: the optional slug for this team. If not provided it will be auto
+                             generated from the name.
+        :qparam bool set_admin: If this is true, the user is added to the as a Team Admin
+                                instead of regular member
         :auth: required
         """
         serializer = TeamPostSerializer(data=request.data)
 
-        if serializer.is_valid():
-            result = serializer.validated_data
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
+        result = serializer.validated_data
+        set_admin = result.get("set_admin")
+
+        if set_admin:
+            if not features.has("organizations:team-roles", organization):
+                return Response(
+                    {
+                        "detail": "Unable to set user as Team Admin because organization:team-roles flag is not enabled"
+                    },
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+            if not features.has("organizations:team-project-creation-all", organization):
+                return Response(
+                    {
+                        "detail": "Unable to set user as Team Admin because organization:team-project-creation-all flag is not enabled"
+                    },
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+            if not self.should_add_creator_to_team(request):
+                return Response(
+                    {
+                        "detail": "Unable to set user as Team Admin because user is not authenticated"
```

Comment:
hm i'm not a huge fan of this wording... maybe something like `You must be authenticated to join a new team as a team admin` or `You are not authenticated to perform this action`